crates/chia-tools/src/bin/benchmark-condition-costs.rs

```diff
@@ -0,0 +1,147 @@
+
+// const SECRET_KEY: &[u8; 32] =
+//     &hex!("6fc9d9a2b05fd1f0e51bc91041a03be8657081f272ec281aff731624f0d1c220");
+
+const MSG1: &[u8; 13] = &[3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3];
+
+// const MSG2: &[u8; 19] = &[4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4];
+
+// this function takes a NodePtr of (q . ((CONDITION ARG ARG)...))
+// and add another (CONDITION ARG ARG) to the list
+fn cons_condition(allocator: &mut Allocator, current_ptr: NodePtr) -> Result<NodePtr, EvalErr> {
+    let Some((q, conds)) = allocator.next(current_ptr) else {
+        return Err(EvalErr(current_ptr, "not a pair".into()));
+    };
+    let Some((cond, rest)) = allocator.next(conds) else {
+        return Err(EvalErr(current_ptr, "not a pair".into()));
+    };
+    let added_new_cond = allocator.new_pair(cond, conds)?;
+    allocator.new_pair(q, added_new_cond)
+}
+
+// this function generates (q . ((CONDITION ARG ARG)))
+fn create_conditions(
+    allocator: &mut Allocator,
+    condition: &ConditionTest,
+) -> Result<NodePtr, EvalErr> {
+    let mut rest = allocator.nil();
+    for arg in condition.args.iter().rev() {
+        rest = allocator.new_pair(*arg, rest)?;
+    }
+    let opcode = allocator.new_small_number(condition.opcode as u32)?;
+    let cond_list = allocator.new_pair(opcode, rest)?;
+    let q = allocator.new_small_number(1)?;
+    let cond_list = allocator.new_pair(cond_list, allocator.nil())?;
+    allocator.new_pair(q, cond_list)
+}
+
+pub fn main() {
+    let mut allocator = Allocator::new();
+    let mut total_cost = 0;
+    let mut total_count = 0;
+    // let puzzle = allocator.new_small_number(1).expect("number");
+    let flags: u32 = COST_CONDITIONS;
+    let mut ret = SpendBundleConditions::default();
+    let mut state = ParseState::default();
+
+    let cond_tests = [
+        ConditionTest {
+            opcode: opcodes::AGG_SIG_UNSAFE,
+            args: vec![
+                allocator.new_atom(PUBKEY).expect("pubkey"),
+                allocator.new_atom(MSG1).expect("msg"),
+            ],
+        },
+        ConditionTest {
+            opcode: opcodes::CREATE_COIN,
+            args: vec![
+                allocator.new_atom(H1).expect("atom"),
+                allocator.new_small_number(1).expect("number"),
+            ],
+        },
+    ];
+    for cond in cond_tests.iter() {
+        let parent_id = allocator.new_atom(H1).expect("atom");
+        let puzzle_hash =
+            Bytes32::from(clvm_utils::tree_hash_from_bytes(&[1_u8]).expect("tree_hash"));
+        let puz_hash_node_ptr = allocator.new_atom(puzzle_hash.as_slice()).expect("bytes");
+        let coin = Coin {
+            parent_coin_info: H1.into(),
+            puzzle_hash: puzzle_hash,
+            amount: 1,
+        };
+        let mut spend = SpendConditions::new(
+            parent_id,
+            1_u64,
+            puz_hash_node_ptr,
+            Arc::new(coin.coin_id()),
+        );
```

Comment:
I think you could hoist this out of the loop. Then you can create a checkpoint of the allocator to reset to in between each test. Right now the allocator grows indefinitely